proxy/deploy.go

```diff
@@ -21,10 +21,25 @@ type FunctionResourceRequest struct {
 	Requests *stack.FunctionResources
 }
 
-// DeployFunction call FaaS server to deploy a new function
 func DeployFunction(fprocess string, gateway string, functionName string, image string,
 	language string, replace bool, envVars map[string]string, network string,
-	constraints []string, update bool, secrets []string, labels map[string]string, functionResourceRequest1 FunctionResourceRequest) {
+	constraints []string, update bool, secrets []string, labels map[string]string,
+	functionResourceRequest1 FunctionResourceRequest) {
+
+	statusCode := Deploy(fprocess, gateway, functionName, image, language, replace, envVars, network, constraints, update, secrets, labels, functionResourceRequest1)
+
+	if update == true && statusCode == http.StatusNotFound {
```

Comment:
Just a *nit* here - can we avoid setting `update = false` and just pass `false` through the message signature? It will make debugging easier. Or you could create a new local variable as an alternative: `updateFunction := false`

Just to avoid destructively updating the value